app/v1/controllers/requestParsers/RetrieveForeignPropertyBISSRequestDataParser.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers
+
+import java.time.LocalDate
+
+import javax.inject.Inject
+import uk.gov.hmrc.domain.Nino
+import utils.DateUtils
+import v1.controllers.requestParsers.validators.RetrieveForeignPropertyBISSValidator
+import v1.models.des.IncomeSourceType
+import v1.models.requestData.{RetrieveForeignPropertyBISSRawData, RetrieveForeignPropertyBISSRequest}
+
+class RetrieveForeignPropertyBISSRequestDataParser @Inject()(val validator: RetrieveForeignPropertyBISSValidator)
+extends RequestParser[RetrieveForeignPropertyBISSRawData, RetrieveForeignPropertyBISSRequest]{
+
+  override protected def requestFor(data: RetrieveForeignPropertyBISSRawData): RetrieveForeignPropertyBISSRequest = {
+   RetrieveForeignPropertyBISSRequest(Nino(data.nino),
+     data.businessId.get,
+     (data.typeOfBusiness: @unchecked) match {
+       case Some("foreign-property") => IncomeSourceType.`foreign-property`
+       case Some("fhl-property-eea") =>  IncomeSourceType.`foreign-property-fhl-eea`
```

Comment:
I think this mapping is back-to-front as foreign-property-fhl-eea is an input to our API and needs to be mapped to fhl-property-eea for DES. I think this means that IncomeSourceType is also wrong.